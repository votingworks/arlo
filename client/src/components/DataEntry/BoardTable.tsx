import React from 'react'
import styled from 'styled-components'
import { H1 } from '@blueprintjs/core'
import { Link } from 'react-router-dom'
import { Column } from 'react-table'
import { Table } from '../Atoms/Table'
import { IAuditBoard, BallotStatus } from '../../types'
import LinkButton from '../Atoms/LinkButton'
import { IBallot } from '../MultiJurisdictionAudit/RoundManagement/useBallots'

const RightWrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  margin: 20px 0;
  .bp3-button {
    margin-left: 10px;
  }
  @media (max-width: 775px) {
    .bp3-button {
      width: 100%;
    }
  }
`

interface IProps {
  boardName: IAuditBoard['name']
  ballots: IBallot[]
  url: string
}

const BoardTable: React.FC<IProps> = ({ boardName, ballots, url }: IProps) => {
  const columns: Column<IBallot>[] = [
    {
      Header: 'Batch',
      accessor: ({ batch: { name } }) => name,
    },
    {
      Header: 'Ballot Position',
      accessor: 'position',
    },
    {
      Header: 'Status',
      // eslint-disable-next-line react/display-name
      accessor: ballot => {
        return ballot.status !== BallotStatus.NOT_AUDITED ? (
          <>
            {ballot.status === BallotStatus.AUDITED ? (
              <span>Audited</span>
            ) : (
              <span>Not Found</span>
            )}
            <Link
              to={`${url}/batch/${ballot.batch.id}/ballot/${ballot.position}`}
              className="bp3-button bp3-small"
            >
              Re-audit
            </Link>
          </>
        ) : (
          'Not Audited'
        )
      },
    },
  ]
  if (ballots.length && ballots[0].batch.tabulator)
    columns.unshift({
      Header: 'Tabulator',
      accessor: ({ batch: { tabulator } }) => tabulator,
    })
  if (ballots.length && ballots[0].batch.container)
    columns.unshift({
      Header: 'Container',
      accessor: ({ batch: { container } }) => container,
    })

  const roundComplete = ballots.every(
    b => b.status !== BallotStatus.NOT_AUDITED
  )

  const unauditedBallot = ballots.find(
    b => b.status === BallotStatus.NOT_AUDITED
  )

  return (
    <div className="board-table-container">
      <H1>{boardName}: Ballot Cards to Audit</H1>
      <p>
        The following ballots have been assigned to your audit board for this
        round of the audit. Once these ballots have been located and retrieved
        from storage, click &quot;Start Auditing&quot; to begin recording the
        votes you see marked on the paper ballots. When you are finished
        auditing these ballots, click &quot;Auditing Complete - Submit
        Results&quot; to submit the results.{' '}
        <strong>
          Note that you will not be able to make changes after results are
          submitted.
        </strong>
      </p>
      <RightWrapper>
        <LinkButton
          to={
            unauditedBallot
              ? `${url}/batch/${unauditedBallot.batch.id}/ballot/${unauditedBallot.position}`
              : ''
          }
          disabled={roundComplete}
        >
          Start Auditing
        </LinkButton>
        <LinkButton to={`${url}/signoff`} disabled={!roundComplete}>
          Auditing Complete - Submit Results
        </LinkButton>
      </RightWrapper>
      {/* <ActionWrapper> // commented out until feature is added
        {!roundComplete && (
          <>
            <Button intent="primary">Download Ballot List as CSV</Button>
          </>
        )}
      </ActionWrapper> */}
      <Table data={ballots} columns={columns} />
    </div>
  )
}

export default BoardTable
