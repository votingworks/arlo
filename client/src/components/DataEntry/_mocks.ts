import { IAuditBoard, BallotStatus, Interpretation } from '../../types'
import { contestMocks } from '../MultiJurisdictionAudit/Setup/Contests/_mocks'
import { IBallot } from './Ballot'

export const contest = contestMocks.filledTargeted.contests[0]

export const doneDummyBallots: { ballots: IBallot[] } = {
  ballots: [
    {
      id: 'ballot-id-1',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-2',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-3',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-4',
      batch: {
        name: '0003-04-Precinct 31 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 1965,
      status: BallotStatus.NOT_FOUND,
      interpretations: [],
    },
  ],
}

export const dummyBallots: { ballots: IBallot[] } = {
  ballots: [
    {
      id: 'ballot-id-1',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: null,
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [
        {
          contestId: contest.id,
          interpretation: Interpretation.VOTE,
          choiceIds: [contest.choices[0].id],
          comment: 'Good ballot',
        },
      ],
    },
    {
      id: 'ballot-id-2',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-3',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-4',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-2',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-5',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-2',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-6',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-2',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-7',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-3',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-8',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-3',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-9',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-3',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-10',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-4',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-11',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-4',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-12',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-4',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-13',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-5',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-14',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-5',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-15',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-5',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-16',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-6',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-17',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-6',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-18',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-6',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-19',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-7',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-20',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-7',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-21',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-7',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-22',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-8',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-23',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-8',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-24',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-8',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-25',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-9',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-26',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-9',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-27',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-9',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
  ],
}

export const dummyBoards = (): IAuditBoard[] => [
  {
    id: 'audit-board-1',
    name: 'Audit Board #1',
    jurisdictionId: 'jurisdiction-1',
    jurisdictionName: 'Jurisdiction 1',
    roundId: 'round-1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    signedOffAt: null,
  },
  {
    id: 'audit-board-2',
    name: 'Audit Board #2',
    jurisdictionId: 'jurisdiction-1',
    jurisdictionName: 'Jurisdiction 1',
    roundId: 'round-1',
    members: [],
    signedOffAt: null,
  },
]

export default {
  dummyBoards,
  dummyBallots,
}
