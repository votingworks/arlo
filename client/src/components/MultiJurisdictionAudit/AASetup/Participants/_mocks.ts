import { readFileSync } from 'fs'
import { join } from 'path'

const jurisdictionFile = new File(
  [
    readFileSync(
      join(
        __dirname,
        '../../../../../public/sample_jurisdiction_filesheet.csv'
      ),
      'utf8'
    ),
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
