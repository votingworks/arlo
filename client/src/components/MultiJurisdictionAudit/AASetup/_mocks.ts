import { ElementType } from '../../../types'
import { setupStages, stageTitles } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
): {
  prevStage: ISidebarMenuItem
  nextStage: ISidebarMenuItem
  menuItems: ISidebarMenuItem[]
} => {
  const prevId = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    id: prevId,
    title: stageTitles[prevId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextId = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    id: nextId,
    title: stageTitles[nextId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const menuItems = setupStages.map((s: ElementType<typeof setupStages>) => ({
    id: s,
    title: stageTitles[s],
    active: s === stage,
    activate: jest.fn(),
    state,
  }))
  return { prevStage, nextStage, menuItems }
}

export default relativeStages
