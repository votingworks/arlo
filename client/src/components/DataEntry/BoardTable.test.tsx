import React from 'react'
import { render, screen } from '@testing-library/react'
import { StaticRouter } from 'react-router-dom'
import BoardTable from './BoardTable'
import { doneDummyBallots } from './_mocks'

describe('BoardTable', () => {
  it('enables the submit button when all ballots are done', async () => {
    const { container } = render(
      <StaticRouter>
        <BoardTable
          boardName="Audit Board #1"
          ballots={doneDummyBallots.ballots}
          url="/home"
        />
      </StaticRouter>
    )
    await screen.findByText('Audit Board #1: Ballot Cards to Audit')

    screen.getByText('Not Found')
    expect(screen.getByText('Auditing Complete - Submit Results')).toBeEnabled()

    expect(container).toMatchSnapshot()
  })
})
