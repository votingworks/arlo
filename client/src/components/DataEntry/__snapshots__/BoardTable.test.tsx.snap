// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardTable enables the submit button when all ballots are done 1`] = `
<div>
  <div
    class="board-table-container"
  >
    <h1
      class="bp3-heading"
    >
      Audit Board #1
      : Ballot Cards to Audit
    </h1>
    <p>
      The following ballots have been assigned to your audit board for this round of the audit. Once these ballots have been located and retrieved from storage, click "Start Auditing" to begin recording the votes you see marked on the paper ballots. When you are finished auditing these ballots, click "Auditing Complete - Submit Results" to submit the results.
       
      <strong>
        Note that you will not be able to make changes after results are submitted.
      </strong>
    </p>
    <div
      class="sc-bdVaJa gKZTDn"
    >
      <button
        class="bp3-button bp3-disabled"
        disabled=""
        href="/"
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Start Auditing
        </span>
      </button>
      <button
        class="bp3-button"
        href="/home/signoff"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Auditing Complete - Submit Results
        </span>
      </button>
    </div>
    <div
      class="bp3-table-container bp3-table-selection-enabled sc-htpNat dCryYg"
      tabindex="0"
    >
      <div
        class="bp3-table-quadrant-stack"
      >
        <div
          class="bp3-table-quadrant bp3-table-quadrant-main"
        >
          <div
            class="bp3-table-quadrant-scroll-container"
          >
            <div
              class="bp3-table-top-container"
            >
              <div
                class="bp3-table-column-headers bp3-table-selection-enabled"
              >
                <div
                  style="width: 600px;"
                >
                  <div
                    class="bp3-table-thead bp3-table-column-header-tr"
                    style="transform: translateX(0px); width: 600px;"
                  >
                    <div
                      class="bp3-table-header bp3-table-cell-col-0 bp3-table-column-header-cell"
                      style="flex-basis: 150px; min-width: 0px; width: 150px;"
                    >
                      <div
                        class="bp3-table-column-name"
                        title="Batch"
                      >
                        <div
                          class="bp3-table-column-name-text"
                        >
                          <div
                            class="bp3-table-truncated-text"
                          >
                            Batch
                          </div>
                        </div>
                      </div>
                      <div
                        class="bp3-table-header-content"
                      />
                      <div
                        class="bp3-table-resize-handle-target bp3-table-resize-vertical"
                      >
                        <div
                          class="bp3-table-resize-handle"
                        />
                      </div>
                    </div>
                    <div
                      class="bp3-table-header bp3-table-cell-col-1 bp3-table-column-header-cell"
                      style="flex-basis: 150px; min-width: 0px; width: 150px;"
                    >
                      <div
                        class="bp3-table-column-name"
                        title="Ballot Position"
                      >
                        <div
                          class="bp3-table-column-name-text"
                        >
                          <div
                            class="bp3-table-truncated-text"
                          >
                            Ballot Position
                          </div>
                        </div>
                      </div>
                      <div
                        class="bp3-table-header-content"
                      />
                      <div
                        class="bp3-table-resize-handle-target bp3-table-resize-vertical"
                      >
                        <div
                          class="bp3-table-resize-handle"
                        />
                      </div>
                    </div>
                    <div
                      class="bp3-table-header bp3-table-cell-col-2 bp3-table-column-header-cell"
                      style="flex-basis: 150px; min-width: 0px; width: 150px;"
                    >
                      <div
                        class="bp3-table-column-name"
                        title="Tabulator"
                      >
                        <div
                          class="bp3-table-column-name-text"
                        >
                          <div
                            class="bp3-table-truncated-text"
                          >
                            Tabulator
                          </div>
                        </div>
                      </div>
                      <div
                        class="bp3-table-header-content"
                      />
                      <div
                        class="bp3-table-resize-handle-target bp3-table-resize-vertical"
                      >
                        <div
                          class="bp3-table-resize-handle"
                        />
                      </div>
                    </div>
                    <div
                      class="bp3-table-header bp3-table-last-in-row bp3-table-cell-col-3 bp3-table-column-header-cell"
                      style="flex-basis: 150px; min-width: 0px; width: 150px;"
                    >
                      <div
                        class="bp3-table-column-name"
                        title="Status"
                      >
                        <div
                          class="bp3-table-column-name-text"
                        >
                          <div
                            class="bp3-table-truncated-text"
                          >
                            Status
                          </div>
                        </div>
                      </div>
                      <div
                        class="bp3-table-header-content"
                      />
                      <div
                        class="bp3-table-resize-handle-target bp3-table-resize-vertical"
                      >
                        <div
                          class="bp3-table-resize-handle"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="bp3-table-bottom-container"
              style="height: 160px; width: 600px;"
            >
              <div
                class="bp3-table-quadrant-body-container"
              >
                <div>
                  <div
                    class="bp3-table-body-virtual-client bp3-table-cell-client"
                    style="height: 160px; width: 600px;"
                  >
                    <div
                      class="bp3-table-body-cells"
                    >
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-0 bp3-table-cell-col-0 bp3-table-cell-ledger-even sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 0px; position: absolute; top: 0px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          0003-04-Precinct 13 (Jonesboro Fire Department)
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-0 bp3-table-cell-col-1 bp3-table-cell-ledger-even sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 150px; position: absolute; top: 0px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          313
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-0 bp3-table-cell-col-2 bp3-table-cell-ledger-even sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 300px; position: absolute; top: 0px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          11
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-0 bp3-table-cell-col-3 bp3-table-last-in-row bp3-table-cell-ledger-even sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 450px; position: absolute; top: 0px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          <span>
                            Audited
                          </span>
                          <a
                            class="bp3-button bp3-small"
                            href="/home/batch/batch-id-1/ballot/313"
                          >
                            Re-audit
                          </a>
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-1 bp3-table-cell-col-0 bp3-table-cell-ledger-odd sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 0px; position: absolute; top: 40px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          0003-04-Precinct 19 (Jonesboro Fire Department)
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-1 bp3-table-cell-col-1 bp3-table-cell-ledger-odd sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 150px; position: absolute; top: 40px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          2112
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-1 bp3-table-cell-col-2 bp3-table-cell-ledger-odd sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 300px; position: absolute; top: 40px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          11
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-1 bp3-table-cell-col-3 bp3-table-last-in-row bp3-table-cell-ledger-odd sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 450px; position: absolute; top: 40px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          <span>
                            Audited
                          </span>
                          <a
                            class="bp3-button bp3-small"
                            href="/home/batch/batch-id-1/ballot/2112"
                          >
                            Re-audit
                          </a>
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-2 bp3-table-cell-col-0 bp3-table-cell-ledger-even sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 0px; position: absolute; top: 80px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          0003-04-Precinct 29 (Jonesboro Fire Department)
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-2 bp3-table-cell-col-1 bp3-table-cell-ledger-even sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 150px; position: absolute; top: 80px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          1789
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-2 bp3-table-cell-col-2 bp3-table-cell-ledger-even sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 300px; position: absolute; top: 80px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          11
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-2 bp3-table-cell-col-3 bp3-table-last-in-row bp3-table-cell-ledger-even sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 450px; position: absolute; top: 80px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          <span>
                            Audited
                          </span>
                          <a
                            class="bp3-button bp3-small"
                            href="/home/batch/batch-id-1/ballot/1789"
                          >
                            Re-audit
                          </a>
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-3 bp3-table-cell-col-0 bp3-table-last-in-column bp3-table-cell-ledger-odd sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 0px; position: absolute; top: 120px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          0003-04-Precinct 31 (Jonesboro Fire Department)
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-3 bp3-table-cell-col-1 bp3-table-last-in-column bp3-table-cell-ledger-odd sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 150px; position: absolute; top: 120px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          1965
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-3 bp3-table-cell-col-2 bp3-table-last-in-column bp3-table-cell-ledger-odd sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 300px; position: absolute; top: 120px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          11
                        </div>
                      </div>
                      <div
                        class="bp3-table-cell bp3-table-truncated-cell bp3-table-cell-row-3 bp3-table-cell-col-3 bp3-table-last-in-column bp3-table-last-in-row bp3-table-cell-ledger-odd sc-bwzfXH gVsDMd"
                        style="height: 40px; left: 450px; position: absolute; top: 120px; width: 150px;"
                      >
                        <div
                          class="bp3-table-truncated-text bp3-table-no-wrap-text"
                        >
                          <span>
                            Not Found
                          </span>
                          <a
                            class="bp3-button bp3-small"
                            href="/home/batch/batch-id-1/ballot/1965"
                          >
                            Re-audit
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bp3-table-quadrant bp3-table-quadrant-top"
          style="height: 30px; right: 0px;"
        >
          <div
            class="bp3-table-quadrant-scroll-container"
          >
            <div
              class="bp3-table-top-container"
            >
              <div
                class="bp3-table-column-headers bp3-table-selection-enabled"
              >
                <div
                  style="width: 600px;"
                >
                  <div
                    class="bp3-table-thead bp3-table-column-header-tr"
                    style="transform: translateX(0px); width: 600px;"
                  >
                    <div
                      class="bp3-table-header bp3-table-cell-col-0 bp3-table-column-header-cell"
                      style="flex-basis: 150px; min-width: 0px; width: 150px;"
                    >
                      <div
                        class="bp3-table-column-name"
                        title="Batch"
                      >
                        <div
                          class="bp3-table-column-name-text"
                        >
                          <div
                            class="bp3-table-truncated-text"
                          >
                            Batch
                          </div>
                        </div>
                      </div>
                      <div
                        class="bp3-table-header-content"
                      />
                      <div
                        class="bp3-table-resize-handle-target bp3-table-resize-vertical"
                      >
                        <div
                          class="bp3-table-resize-handle"
                        />
                      </div>
                    </div>
                    <div
                      class="bp3-table-header bp3-table-cell-col-1 bp3-table-column-header-cell"
                      style="flex-basis: 150px; min-width: 0px; width: 150px;"
                    >
                      <div
                        class="bp3-table-column-name"
                        title="Ballot Position"
                      >
                        <div
                          class="bp3-table-column-name-text"
                        >
                          <div
                            class="bp3-table-truncated-text"
                          >
                            Ballot Position
                          </div>
                        </div>
                      </div>
                      <div
                        class="bp3-table-header-content"
                      />
                      <div
                        class="bp3-table-resize-handle-target bp3-table-resize-vertical"
                      >
                        <div
                          class="bp3-table-resize-handle"
                        />
                      </div>
                    </div>
                    <div
                      class="bp3-table-header bp3-table-cell-col-2 bp3-table-column-header-cell"
                      style="flex-basis: 150px; min-width: 0px; width: 150px;"
                    >
                      <div
                        class="bp3-table-column-name"
                        title="Tabulator"
                      >
                        <div
                          class="bp3-table-column-name-text"
                        >
                          <div
                            class="bp3-table-truncated-text"
                          >
                            Tabulator
                          </div>
                        </div>
                      </div>
                      <div
                        class="bp3-table-header-content"
                      />
                      <div
                        class="bp3-table-resize-handle-target bp3-table-resize-vertical"
                      >
                        <div
                          class="bp3-table-resize-handle"
                        />
                      </div>
                    </div>
                    <div
                      class="bp3-table-header bp3-table-last-in-row bp3-table-cell-col-3 bp3-table-column-header-cell"
                      style="flex-basis: 150px; min-width: 0px; width: 150px;"
                    >
                      <div
                        class="bp3-table-column-name"
                        title="Status"
                      >
                        <div
                          class="bp3-table-column-name-text"
                        >
                          <div
                            class="bp3-table-truncated-text"
                          >
                            Status
                          </div>
                        </div>
                      </div>
                      <div
                        class="bp3-table-header-content"
                      />
                      <div
                        class="bp3-table-resize-handle-target bp3-table-resize-vertical"
                      >
                        <div
                          class="bp3-table-resize-handle"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="bp3-table-bottom-container"
              style="height: 160px; width: 600px;"
            >
              <div
                class="bp3-table-quadrant-body-container"
              >
                <div>
                  <div
                    class="bp3-table-body-virtual-client bp3-table-cell-client"
                    style="width: 600px;"
                  >
                    <div
                      class="bp3-table-body-cells"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="bp3-table-overlay-layer bp3-table-reordering-cursor-overlay"
      />
      <div
        class="bp3-table-resize-guides bp3-table-overlay-layer"
      />
      <div
        class="bp3-table-resize-sensor"
        style="position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden;"
      >
        <div
          class="bp3-table-resize-sensor-expand"
          style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
        >
          <div
            style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 100000px; height: 100000px;"
          />
        </div>
        <div
          class="bp3-table-resize-sensor-shrink"
          style="position: absolute; left: 0; top: 0; right: 0; bottom: 0; overflow: hidden; z-index: -1; visibility: hidden;"
        >
          <div
            style="position: absolute; left: 0; top: 0; transition: 0s; width: 200%; height: 200%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
